Module path prefix is not used to resolve styleUrl or templateUrl with AoT

### OS?
> Windows 10


### Versions.
> @angular/cli: 1.0.0-rc.0
node: 7.5.0
os: win32 x64
> @angular/common: 4.0.0-rc.1
> @angular/compiler: 4.0.0-rc.1
> @angular/core: 4.0.0-rc.1
> @angular/flex-layout: 2.0.0-rc.1
> @angular/forms: 4.0.0-rc.1
> @angular/http: 4.0.0-rc.1
> @angular/platform-browser: 4.0.0-rc.1
> @angular/platform-browser-dynamic: 4.0.0-rc.1
> @angular/router: 4.0.0-rc.1
> @angular/cli: 1.0.0-rc.0
> @angular/compiler-cli: 4.0.0-rc.1

### Repro steps.

Due to reasons we have our style sheets in a separate root folder. A couple versions ago we were able to load the styling using:

    @Component({
        //...
        styleUrls: [
            "@style_folder/components/component/name.component.scss"
        ]
    })

Then something changed a couple versions ago and module paths were resolved as relative paths, so `./` was appended to the url. This is done in the `loader-utils` package. I've read there that I could append a `~` to the url, to convert it to a module request. (`~@style_folder/bla/bla`) This worked splendidly (with the minor downside that I could no longer navigate to it using my IDE).

We are in the process of making our application AoT compatible, and to our big surprise, the `~` trick doesn't work there, because AoT thinks it's a relative path no matter what.

We would prefer it to not have all relative paths inside our `styleUrls` (../../../ horror). Better would be that both JIT and AoT look at the paths defined in `tsconfig.json` (like it was before I believe) and look there for a definition, and only after that assume it's a relative path if there is no match.

@PierreDuc Is there a repo that I can check the issue. I tried to replicate and it seems to be failing in bot `JIT` and `AOT` with or witout `~`.
```
 ~/w/o/t/t2   master *…  ng s                                                                                           1.4m  Tue 14 Mar 13:13:38 2017
** NG Live Development Server is running on http://localhost:4200 **
Hash: 789faea35224a98f2cfe
Time: 4933ms
chunk    {0} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 158 kB {4} [initial] [rendered]
chunk    {1} main.bundle.js, main.bundle.js.map (main) 5.06 kB {3} [initial] [rendered]
chunk    {2} styles.bundle.js, styles.bundle.js.map (styles) 9.91 kB {4} [initial] [rendered]
chunk    {3} vendor.bundle.js, vendor.bundle.js.map (vendor) 2.26 MB [initial] [rendered]
chunk    {4} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

ERROR in ./src/app/hello/hello.component.ts
Module not found: Error: Can't resolve '@style-paths/hello.component' in '/Users/sumitarora/workspace/open-source/temp/t2/src/app/hello'
 @ ./src/app/hello/hello.component.ts 22:17-56
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
webpack: Failed to compile.
^C⏎                                                                                                                                                        !  ~/w/o/t/t2   master *…  ng s --aot                                                                                13.3s  Tue 14 Mar 13:21:33 2017
** NG Live Development Server is running on http://localhost:4200 **
Hash: f040d2d99ec59b023a6d
Time: 2439ms
chunk    {0} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 158 kB {4} [initial] [rendered]
chunk    {1} styles.bundle.js, styles.bundle.js.map (styles) 9.91 kB {4} [initial] [rendered]
chunk    {2} main.bundle.js, main.bundle.js.map (main) 1.1 kB {3} [initial] [rendered]
chunk    {3} vendor.bundle.js, vendor.bundle.js.map (vendor) 1.07 MB [initial] [rendered]
chunk    {4} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

ERROR in ENOENT: no such file or directory, open '/Users/sumitarora/workspace/open-source/temp/t2/src/app/hello/~@style-paths/hello.component'

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/Users/sumitarora/workspace/open-source/temp/t2/src'
 @ ./src/main.ts 4:0-74
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
webpack: Failed to compile.
^C⏎                                                                                                                                                        !  ~/w/o/t/t2   master *…  ng s                                                                                      22.6s  Tue 14 Mar 13:21:59 2017
** NG Live Development Server is running on http://localhost:4200 **
Hash: f7a7a95c3f6047402914
Time: 4891ms
chunk    {0} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 158 kB {4} [initial] [rendered]
chunk    {1} main.bundle.js, main.bundle.js.map (main) 5.06 kB {3} [initial] [rendered]
chunk    {2} styles.bundle.js, styles.bundle.js.map (styles) 9.91 kB {4} [initial] [rendered]
chunk    {3} vendor.bundle.js, vendor.bundle.js.map (vendor) 2.26 MB [initial] [rendered]
chunk    {4} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

ERROR in ./src/app/hello/hello.component.ts
Module not found: Error: Can't resolve './@style-paths/hello.component' in '/Users/sumitarora/workspace/open-source/temp/t2/src/app/hello'
 @ ./src/app/hello/hello.component.ts 22:17-58
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
webpack: Failed to compile.
^C⏎                                                                                                                                                        !  ~/w/o/t/t2   master *…  ng s --aot                                                                                23.4s  Tue 14 Mar 13:22:31 2017
** NG Live Development Server is running on http://localhost:4200 **
Hash: d4f674febff20caea8c9
Time: 2440ms
chunk    {0} polyfills.bundle.js, polyfills.bundle.js.map (polyfills) 158 kB {4} [initial] [rendered]
chunk    {1} styles.bundle.js, styles.bundle.js.map (styles) 9.91 kB {4} [initial] [rendered]
chunk    {2} main.bundle.js, main.bundle.js.map (main) 1.1 kB {3} [initial] [rendered]
chunk    {3} vendor.bundle.js, vendor.bundle.js.map (vendor) 1.07 MB [initial] [rendered]
chunk    {4} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

ERROR in ENOENT: no such file or directory, open '/Users/sumitarora/workspace/open-source/temp/t2/src/app/hello/@style-paths/hello.component'

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/Users/sumitarora/workspace/open-source/temp/t2/src'
 @ ./src/main.ts 4:0-74
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
webpack: Failed to compile.
^C⏎
```

--- a/packages/@ngtools/webpack/src/loader.ts
+++ b/packages/@ngtools/webpack/src/loader.ts
@@ -262,8 +262,9 @@ function _removeModuleId(refactor: TypeScriptFileRefactor) {
 
 function _getResourceRequest(element: ts.Expression, sourceFile: ts.SourceFile) {
   if (element.kind == ts.SyntaxKind.StringLiteral) {
-    // if string, assume relative path unless it start with /
-    return `'${loaderUtils.urlToRequest((element as ts.StringLiteral).text, '')}'`;
+    const url = (element as ts.StringLiteral).text;
+    // If the URL does not start with ./ or ../, prepends ./ to it.
+    return `'${/^\.?\.\//.test(url) ? '' : './'}${url}'`;
   } else {
     // if not string, just use expression directly
     return element.getFullText(sourceFile);

